apps/frontend/src/hooks/aave/useAaveRates.tsx

```diff
@@ -0,0 +1,300 @@
+          },
+          {
+            internalType: 'uint256',
+            name: 'liquidityAdded',
+            type: 'uint256',
+          },
+          {
+            internalType: 'uint256',
+            name: 'liquidityTaken',
+            type: 'uint256',
+          },
+          {
+            internalType: 'uint256',
+            name: 'totalVariableDebt',
+            type: 'uint256',
+          },
+          {
+            internalType: 'uint256',
+            name: 'reserveFactor',
+            type: 'uint256',
+          },
+          {
+            internalType: 'address',
+            name: 'reserve',
+            type: 'address',
+          },
+          {
+            internalType: 'address',
+            name: 'aToken',
+            type: 'address',
+          },
+        ],
+        internalType: 'struct DataTypes.CalculateInterestRatesParams',
+        name: 'params',
+        type: 'tuple',
+      },
+    ],
+    name: 'calculateInterestRates',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256',
+      },
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256',
+      },
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    inputs: [],
+    name: 'getBaseVariableBorrowRate',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    inputs: [],
+    name: 'getMaxVariableBorrowRate',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    inputs: [],
+    name: 'getVariableRateSlope1',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    inputs: [],
+    name: 'getVariableRateSlope2',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+];
+
+export interface RatesDataResult {
+  currentUsageRatio: string;
+  optimalUsageRatio: string;
+  baseVariableBorrowRate: string;
+  variableRateSlope1: string;
+  variableRateSlope2: string;
+  underlyingAsset: string;
+  name: string;
+  symbol: string;
+  decimals: string;
+}
+
+const calculateUtilizationRate = (
+  decimals: number,
+  totalDebt: string,
+  availableLiquidity: string,
+): bigint => {
+  // Create BigNumber instances
+  const totalBorrow = BigInt(utils.parseUnits(totalDebt, decimals).toString());
+  const totalSupply = BigInt(availableLiquidity) + totalBorrow;
+  // Perform division
+  return (totalBorrow * BigInt(10 ** 18)) / totalSupply;
+};
```

Comment:
For units we use `Decimal` and `BigNumber` from `ethers`

Reply:
I asked @juandahl  to review the math for this function, as I was using BigNumber (as you pointed out) but was encountering unexpected results, with many cases returning zero. Since I'm not very experienced with this data type, let's work through the solution together. All feedback and suggestions are welcome.